docs/source/entities-advanced.mdx

```diff
@@ -22,10 +48,34 @@ type Product @key(fields: "id") @key(fields: "sku") {
   price: Int
 }
 ```
+<blockquote>
+
+**Note:** If you include multiple sets of `@key` fields, the query planner always uses the most efficient set for entity resolution. For example, suppose you allow a type to be identified by `@key(fields: "id")` _or_ `@key(fields: "id sku")`:
```

Comment:
Copyedit suggestion: is there a more precise definition for `most efficient`? Is it the least number of operands?

Reply:
Good question. That's how I understood it from [this conversation](https://github.com/apollographql/federation/issues/2695), but I wonder what else could be at play here. For example, if two different keys both use the same number of fields, how does the query planner decide which is more efficient? Does it calculate the shortest "path" to resolve fields?

@dariuszkuc , would you be able to help with this?

Reply:
> if two different keys both use the same number of fields, how does the query planner decide which is more efficient? Does it calculate the shortest "path" to resolve fields?



@pcmanus by any chance do you know whats the optimization logic here?

Reply:
Got a chance to check in with @pcmanus and learned that the query planner a cost function to score keys, and it usually selects the key with the fewest fields in case of ties, although there are no strict guarantees. Since this is likely to come up infrequently, I'm going to leave it relatively vague.